src/Games/NexusMods.Games.StardewValley/Emitters/SMAPIGameVersionDiagnosticEmitter.cs

```diff
@@ -0,0 +1,328 @@
+        if (_smapiToGameMappings is not null) return _smapiToGameMappings;
+
+        var data = await FetchData<Dictionary<string, GameVersionsDTO>>(SMAPIToGameMappingsDataUri, cancellationToken);
+        if (data is null) return null;
+
+        var res = new Dictionary<Version, GameVersions>();
+        foreach (var kv in data)
+        {
+            var (sVersion, gameVersions) = kv;
+            var (sMinimumGameVersion, sMaximumGameVersion) = gameVersions;
+
+            if (!TryParseVersion(sVersion, out var version)) continue;
+
+            Version? minimumGameVersion = null, maximumGameVersion = null;
+            if (sMinimumGameVersion is not null) if (!TryParseVersion(sMinimumGameVersion, out minimumGameVersion)) continue;
+            if (sMaximumGameVersion is not null) if (!TryParseVersion(sMaximumGameVersion, out maximumGameVersion)) continue;
+
+            res[version] = new GameVersions
+            {
+                MinimumGameVersion = minimumGameVersion,
+                MaximumGameVersion = maximumGameVersion,
+            };
+        }
+
+        _smapiToGameMappings = res.ToImmutableDictionary(keyComparer: VersionComparer.Instance);
+        return _smapiToGameMappings;
+
+    }
+
+    private async Task<GameToSMAPIMapping?> FetchGameToSMAPIMappings(CancellationToken cancellationToken)
+    {
+        if (_gameToSMAPIMappings is not null) return _gameToSMAPIMappings;
+
+        var data = await FetchData<Dictionary<string, string[]>>(GameToSMAPIMappingsDataUri, cancellationToken);
+        if (data is null) return null;
+
+        var res = new Dictionary<Version, Version>();
+        foreach (var kv in data)
+        {
+            var (sGameVersion, smapiVersions) = kv;
+            if (smapiVersions.Length == 0) continue;
+
+            if (!TryParseVersion(sGameVersion, out var gameVersion)) continue;
+
+            var sSMAPIVersion = smapiVersions.First();
+            if (!TryParseVersion(sSMAPIVersion, out var smapiVersion)) continue;
+
+            res[gameVersion] = smapiVersion;
+        }
+
+        _gameToSMAPIMappings = res.ToImmutableDictionary(keyComparer: VersionComparer.Instance);
+        return _gameToSMAPIMappings;
+    }
+
+    private bool TryParseVersion(string input, [NotNullWhen(true)] out Version? version)
+    {
+        version = null;
+        if (Version.TryParse(input, out var tmp))
+        {
+            version = SimplifyVersion(tmp);
+            return true;
+        }
+
+        _logger.LogWarning("Unable to parse string `{Input}` as a Version", input);
+        return false;
+    }
+
+    private static Version SimplifyVersion(Version input)
+    {
+        var major = input.Major;
+        var minor = input.Minor;
+        var build = input.Build == -1 ? 0 : input.Build;
+        var revision = input.Revision == 0 ? -1 : input.Revision;
+
+        if (revision != -1) return new Version(major, minor, build, revision);
+        return new Version(major, minor, build);
+    }
```

Comment:
I'm not sure what the original intent for this function was, but I tested always stripping out the `revision` number and that fixed the issues with version checks being messed up. The version numbers in the description are also stripped out that way, but it arguably makes things easier since revision number isn't really relevant in this case.